tide/core/DisplayGroup.cpp

```diff
@@ -297,33 +245,37 @@ ContentWindow* DisplayGroup::getFullscreenWindow() const
     return _fullscreenWindow.get();
 }
 
-void DisplayGroup::setShowWindowTitles( const bool set )
+void DisplayGroup::setFullscreenWindow( ContentWindowPtr window )
 {
-    if( _showWindowTitlesInSavedSession == set )
+    if( _fullscreenWindow == window )
         return;
 
-    _showWindowTitlesInSavedSession = set;
-}
-
-void DisplayGroup::clear()
-{
-    if( _contentWindows.empty( ))
-        return;
+    // restore window state
+    if( _fullscreenWindow )
+    {
+        _fullscreenWindow->setMode( _fullscreenWindowPrevMode );
+        _fullscreenWindow->getContent()->setZoomRect( _fullscreenWindowPrevZoom );
+    }
 
-    // Close regular windows but hide panels (instead of removing them)
-    ContentWindowPtrs removeSet;
-    for( auto window : _contentWindows )
+    // backup window state
+    if( window )
     {
-        if( window->isPanel( ))
-            window->setState( ContentWindow::HIDDEN );
-        else
-            removeSet.push_back( window );
+        _fullscreenWindowPrevMode = window->getMode();
+        _fullscreenWindowPrevZoom = window->getContent()->getZoomRect();
+        window->setMode( ContentWindow::WindowMode::FULLSCREEN );
     }
 
-    put_flog( LOG_INFO, "removing %i windows", removeSet.size( ));
+    _fullscreenWindow = window;
+    emit hasFullscreenWindowsChanged();
+    _sendDisplayGroup();
+}
 
-    for( auto window : removeSet )
-        removeContentWindow( window );
+void DisplayGroup::setShowWindowTitles( const bool set )
+{
+    if( _showWindowTitlesInSavedSession == set )
```

Comment:
this early doesn't save anything, no? _showWindowTitlesInSavedSession is a plain bool